src/lib.rs

```diff
@@ -116,15 +120,37 @@ where
     }
 }
 
+/// If "ark-serialize" feature enabled
+/// then add `CanonicalSerialize`, `CanonicalDeserialize`
+/// else add nothing
+#[cfg(feature = "ark-serialize")]
+pub trait CommitmentBoundsSerdeless:
+    CommitmentBoundsArkless + CanonicalDeserialize + CanonicalSerialize
+{
+}
+
+#[cfg(not(feature = "ark-serialize"))]
+pub trait CommitmentBoundsSerdeless: CommitmentBoundsArkless {}
+
+impl<T> CommitmentBoundsSerdeless for Commitment<T> where T: Committable + 'static {}
+
+/// If "serde" feature enabled
+/// then add `Serialize`, `Deserialize`
+/// else if "ark-serialize" enabled but not "serde"
+/// then add nothing to [`CommitmentBoundsSerdeless`]
+/// else add nothing to [`CommitmentBoundsArkless`]
```

Comment:
Why can't we also use `CommitmentBoundsSerdeless` in this case? Since if ark-serialize is not enabled, `CommitmentBoundsSerdeless` is defined to `CommitmentBoundsArkless`?

Reply:
Good catch. I must have been wearing my stupid hat that day. Fixed in https://github.com/EspressoSystems/commit/pull/40/commits/4b0f53bdf58422eef7176632e5b05d71b1e5de44 .